pandas/core/strings.py

```diff
@@ -464,11 +464,9 @@ def rep(x, r):
         return result
 
 
-def str_match(arr, pat, case=True, flags=0, na=np.nan, as_indexer=False):
+def str_match(arr, pat, case=True, flags=0, na=np.nan, as_indexer=None):
```

Comment:
well its going to be removed. So should for sure use ``FutureWarning``. ``UserWarning`` is pretty useless as a warning IMHO. (not that ``FutureWarning`` is much better but at least signals that we are going to remove it).